### PR TITLE
Home work 3

### PR DESCRIPTION
**Pull request by YaroslavO**

### CODE REVIEW

**Comment by powerbot15** [comment]
Линтер ругается, надо поисправлять
```
homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.css
 220:1   ✖  Expected empty line before rule       rule-empty-line-before       
 223:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 232:1   ✖  Expected empty line before rule       rule-empty-line-before       
 239:1   ✖  Expected empty line before rule       rule-empty-line-before       
 247:1   ✖  Expected empty line before rule       rule-empty-line-before       
 249:11  ✖  Unexpected leading zero               number-leading-zero          
 258:1   ✖  Expected empty line before rule       rule-empty-line-before       
 261:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 263:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 271:1   ✖  Expected empty line before rule       rule-empty-line-before       
 274:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 283:1   ✖  Expected empty line before rule       rule-empty-line-before       
 291:1   ✖  Expected empty line before rule       rule-empty-line-before       
 294:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 296:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 303:1   ✖  Expected empty line before rule       rule-empty-line-before       
 305:16  ✖  Expected numeric font-weight          font-weight-notation         
            notation                                                           
 312:1   ✖  Expected empty line before rule       rule-empty-line-before       
 317:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 324:1   ✖  Expected empty line before rule       rule-empty-line-before       
 332:1   ✖  Expected empty line before rule       rule-empty-line-before       
 341:12  ✖  Unexpected leading zero               number-leading-zero          
 345:9   ✖  Unexpected leading zero               number-leading-zero          
 353:1   ✖  Expected empty line before rule       rule-empty-line-before       
 363:1   ✖  Expected empty line before rule       rule-empty-line-before       
 370:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 372:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 374:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 382:1   ✖  Expected empty line before rule       rule-empty-line-before       
 385:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 392:1   ✖  Expected empty line before rule       rule-empty-line-before       
 395:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 401:1   ✖  Expected empty line before rule       rule-empty-line-before       
 411:1   ✖  Expected empty line before rule       rule-empty-line-before       
 422:1   ✖  Expected empty line before rule       rule-empty-line-before       
 432:1   ✖  Expected empty line before rule       rule-empty-line-before       
 433:12  ✖  Unexpected leading zero               number-leading-zero          
 433:19  ✖  Unexpected leading zero               number-leading-zero          
 433:26  ✖  Unexpected leading zero               number-leading-zero          
 442:1   ✖  Expected empty line before rule       rule-empty-line-before       
 445:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 447:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 449:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 451:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 453:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 460:1   ✖  Expected empty line before rule       rule-empty-line-before       
 467:1   ✖  Expected empty line before rule       rule-empty-line-before       
 475:1   ✖  Expected empty line before rule       rule-empty-line-before       
 479:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 486:1   ✖  Expected empty line before rule       rule-empty-line-before       
 495:1   ✖  Expected empty line before rule       rule-empty-line-before       
 498:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 505:1   ✖  Expected empty line before rule       rule-empty-line-before       
 513:1   ✖  Expected empty line before rule       rule-empty-line-before       
 516:3   ✖  Expected empty line before            declaration-empty-line-before
            declaration                                                        
 525:1   ✖  Expected empty line before rule       rule-empty-line-before       
 532:1   ✖  Expected empty line before rule       rule-empty-line-before       
 541:1   ✖  Expected empty line before rule       rule-empty-line-before       
 548:1   ✖  Expected empty line before rule       rule-empty-line-before

```

**Comment by YaroslavO** [comment]
> Линтер ругается, надо поисправлять
> 
> ```
> homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.css
>  220:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  223:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  232:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  239:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  247:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  249:11  ✖  Unexpected leading zero               number-leading-zero          
>  258:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  261:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  263:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  271:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  274:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  283:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  291:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  294:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  296:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  303:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  305:16  ✖  Expected numeric font-weight          font-weight-notation         
>             notation                                                           
>  312:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  317:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  324:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  332:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  341:12  ✖  Unexpected leading zero               number-leading-zero          
>  345:9   ✖  Unexpected leading zero               number-leading-zero          
>  353:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  363:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  370:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  372:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  374:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  382:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  385:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  392:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  395:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  401:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  411:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  422:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  432:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  433:12  ✖  Unexpected leading zero               number-leading-zero          
>  433:19  ✖  Unexpected leading zero               number-leading-zero          
>  433:26  ✖  Unexpected leading zero               number-leading-zero          
>  442:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  445:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  447:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  449:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  451:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  453:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  460:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  467:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  475:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  479:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  486:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  495:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  498:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  505:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  513:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  516:3   ✖  Expected empty line before            declaration-empty-line-before
>             declaration                                                        
>  525:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  532:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  541:1   ✖  Expected empty line before rule       rule-empty-line-before       
>  548:1   ✖  Expected empty line before rule       rule-empty-line-before
> ```

Я использовал scss. поэтому не стал заморачиваться над отступами.
Чтоб эти ошибки исчезли нужно убрать normalize.css 
Линтер не понимает что код сгенерирован.

**Comment by TArch64** [comment]
Просто підключи нормалайз через CDN 
```html
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css" integrity="sha256-l85OmPOjvil/SOvVt3HnSSjzF1TUMyT9eV0c2BzEGzU=" crossorigin="anonymous" />
```

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const form = document.getElementsByClassName('article__form')[0];
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js]: Просто юзай `querySelector`. З ним простіше

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+const name = document.getElementsByClassName('article__input-name')[0];
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js]: Селектор для js має бути унікальним і до таких селекторів краще додавати префікс js щоб потім при рефакторингу стилів знати що це в js використовується і не поломати нічого

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+const email = document.getElementsByClassName('article__input-email')[0];
+const text = document.getElementsByClassName('article__textarea')[0];
+
+form.onsubmit = (event) => {
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js]: Використовуй краще addEventListener. там більше опшинів і в випадку чого можна повісити декілька колбеків на один івент

File: homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+</head>
+<body>
+<div class="progress-bar" data-progress></div>
+<img style="max-width: 100%;" data-cat />
+
+<script defer>
+    (() => {
+        class HTTPClient {
+            get(url, options = {}) {
+                const {responseType = '', onprogress, method = "GET"} = options;
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html]: Навіщо method передавати? у тебе тут окремий метод для гет запита і ти завжди знаєш що method = "GET"

File: homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                const xhr = new XMLHttpRequest();
+
+                return new Promise((resolve) => {
+                    xhr.onprogress = onprogress;
+                    xhr.responseType = responseType;
+
+                    xhr.onload = () => {
+                        resolve({body: xhr.response});
+                    };
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html]: xhr.onload = () => resolve({body: xhr.response});

File: homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+                    xhr.open(method, url);
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html:43]: Тут ти завжди знаєш, що в method лежить `'GET'`. Можна навіть літералом забити

File: homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html (hunk 4 above)
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html]: В responseType не може бути пустої строки. Там завжди якесь значення. Частіше всього юзеру потрібно буде там `application/json`

**Review by TArch64 — CHANGES_REQUESTED** [review]

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js (hunk 1 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js]: +

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js (hunk 2 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js]: +

File: homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html (hunk 5 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html]: +

File: homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html (hunk 4 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html]: Я не знав що по дефолту треба "application/json"
В доці написано
""
An empty responseType string is treated the same as "text", the default type.
Тому вирішив порожню строку.

Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html]: Просто хотів крутіший код написати)

File: homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html (hunk 6 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html:43]: +

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js (hunk 1 above)
Comment (TArch64) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js]: Що то за точечка?

File: homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html (hunk 4 above)
Comment (TArch64) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html]: Ну так зазвичай щоб було крутіше, то роблять клас, а в ньому методи get, post, patch, delete etc як відповідають за певний http метод

Comment (TArch64) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_3/index.html]: В докі написано про text, але зараз на дворі 2019 рік і скоріш всього всі люди будуть ставити application/json. Бо текстом чи xml зараз ніхто і нічого не пересилає. Люди майже завжди використовують json. Думаю можна навіть зробити константу DEFAULT_RESPONSE_TYPE де прописати людський тип респонсу, якщо ніхто нічого не передав

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        <a class="menu__item menu__item_active" href="/">Home</a>
+        <a class="menu__item" href="/">About</a>
+        <a class="menu__item" href="/">Service</a>
+        <a class="menu__item" href="/">Resume</a>
+        <a class="menu__item" href="/">Portfolio</a>
+        <a class="menu__item" href="/">Blog</a>
+        <a class="menu__item" href="/">Contact Us</a>
+    </div>
+</nav>
+<section class="articles__section">
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: З семантичної точки зору це ні разу не section. Більше сюди підходить main, бо в тебе тут тільки одна секція, в якій весь контент сторінки

Comment (TArch64) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Якщо дивитись з точки зору БЕМ на `articles__section`, то в цей клас є елементом `articles`, але останнього в тебе немає. Назви якось по іншому клас. щось типу sections

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+    <div class="menu__group">
+        <a class="menu__item menu__item_active" href="/">Home</a>
+        <a class="menu__item" href="/">About</a>
+        <a class="menu__item" href="/">Service</a>
+        <a class="menu__item" href="/">Resume</a>
+        <a class="menu__item" href="/">Portfolio</a>
+        <a class="menu__item" href="/">Blog</a>
+        <a class="menu__item" href="/">Contact Us</a>
+    </div>
+</nav>
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: `menu ` по фактом є header сторінки. і трохи не навігацією (лого може не входити в тег навігації). А отой `menu__group ` уже є том самим nav

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+                Hello,
+            </div>
+            <div class="article__first_greeting article__first_text">
+                I am John Smith
+            </div>
+            <div class="article__first_text">
+                Freelance web Designer
+                <br/>
+                @ Developer
+            </div>
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Для тексту є тег p

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+        <a class="menu__item" href="/">Service</a>
+        <a class="menu__item" href="/">Resume</a>
+        <a class="menu__item" href="/">Portfolio</a>
+        <a class="menu__item" href="/">Blog</a>
+        <a class="menu__item" href="/">Contact Us</a>
+    </div>
+</nav>
+<section class="articles__section">
+    <article class="article article__first">
+        <div class="article__content_left">
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Поменше використовуй назви left right, а first, next, second старайся взагалі не використоувати. Замість таких слів напиши назву, яка буде говорити про контент

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+            <div class="article__first_text">
+                Hello,
+            </div>
+            <div class="article__first_greeting article__first_text">
+                I am John Smith
+            </div>
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Тут у тебе якась магія з 3ма колонками. Зроби просто 2 колонки і щоб перша була в 2 рази більше другої
image

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+        <a class="menu__item" href="/">About</a>
+        <a class="menu__item" href="/">Service</a>
+        <a class="menu__item" href="/">Resume</a>
+        <a class="menu__item" href="/">Portfolio</a>
+        <a class="menu__item" href="/">Blog</a>
+        <a class="menu__item" href="/">Contact Us</a>
+    </div>
+</nav>
+<section class="articles__section">
+    <article class="article article__first">
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: З семантичної точки зору це не article. це є section, бо це частина сторінки, article більше про окрему сутність (наприклад пост в соц мережі)

Comment (TArch64) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: І не використовуй назви тегів в селекторах. Ти прив'язуєш використання до певних тегів. в один момент вони молжуть (і скоріше всього) зміняться на інші і це буде плутати людей, або хтось може не використовувати цей клас, бо буде думати що він тільки для цього тегу працює, а там просто падінги і фон

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+        <div class="article__content_left">
+            <div class="article__first_text">
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Замість того щоб писати кожен раз новий клас з такими страшними назвами зроби собі класи типу text--regular, text--heading, text--subheading etc, і для кольорів text--primary, text--secondary, text--warning etc

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js (hunk 1 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js]: це так гітхаб плюс перетворює

**Review by TArch64 — CHANGES_REQUESTED** [review]

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js (hunk 1 above)
Comment (TArch64) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js]: Замість точки нажимай кнопку 
image

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html (hunk 7 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Часу думати не було, тому так понаписував.

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html (hunk 9 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Я його не брав бо він додавав margin.

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html (hunk 10 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Знову ж таки все впирається в час

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html (hunk 11 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Логічно. Теж так думав

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html (hunk 12 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: це я один article пропустив)

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html (hunk 13 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Логічно. Теж так думав. Написав як швидше

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html (hunk 9 above)
Comment (TArch64) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Ну так маржін можна прибрати. Оберай теги не по стилях, які на них висять, а по тому для чого вони вони використовуються

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html (hunk 7 above)
Comment (TArch64) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Ми тут не сайтец по дедлайнах для замовників пилимо. А пробуємо вас навчити і показати як потрібно. А в цьому сенсу немає, якщо всі будуть робити як небуть, бо немає часу

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+        <a class="menu__item menu__item_active" href="/">Home</a>
+        <a class="menu__item" href="/">About</a>
+        <a class="menu__item" href="/">Service</a>
+        <a class="menu__item" href="/">Resume</a>
+        <a class="menu__item" href="/">Portfolio</a>
+        <a class="menu__item" href="/">Blog</a>
+        <a class="menu__item" href="/">Contact Us</a>
+    </div>
+</nav>
+<section class="articles__section">
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: `<section class="articles__section">` тут у тебе не секція, а основний контент сторінки в якого є різні секції. Для такого зазвичай використовують main

Comment (TArch64) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: articles__section І цей клас каже що у тебе тут є блок articles, а це його елемент (частина блоку, яка без нього не може існувати), але тут у тебе немає блоку articles

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+    <section class="article article__first">
+        <div class="article__content_left">
+            <div class="article__first_text">
+                Hello,
+            </div>
+            <div class="article__first_greeting article__first_text">
+                I am John Smith
+            </div>
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: використовуй класи, які для цього робили та переназивай класи за тим як вони використовуються.

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+        </header>
+        <section class="article__content_right">
+            <form class="article__form">
+                <input type="text" placeholder="Your Name" class="article__input article__input-name" required pattern="[A-z]{2,}\s[A-z]{2,}">
+                <input type="email" placeholder="Email Address" class="article__input article__input-email" pattern="\S+@\S+\.\S+" required>
+                <textarea placeholder="Your Messages" class="article__input article__textarea" required></textarea>
+                <button type="submit" class="button action-button">Send Message</button>
+            </form>
+        </section>
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: У тебе тут секція всередині секції. Щось з цим не так. І я трохи сумнівать про необхідніть <section class="article__content_right">. Скоріш вього він тут не потрібен

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html [hunk 17]
```diff
@@ -0,0 +1,122 @@
...
+    </section>
+</section>
+<footer>
+
+</footer>
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Навіщо тобі пустий футер?

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html [hunk 18]
```diff
@@ -0,0 +1,122 @@
...
+
+<script src="index.js"></script>
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: додавай до скріптів defer, щоб він відпрацьовував після відмальовки дому. Без нього іноді можуть не знаходитись елементи в домі

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js [hunk 19]
```diff
@@ -0,0 +1,11 @@
+const getElement = (selector) => document.querySelector(selector);
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js:1]: Навіщо так тяжко з getElement. Просто заюзай document.querySelector. Ти не так то і багато економиш букв

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js [hunk 20]
```diff
@@ -0,0 +1,11 @@
...
+const form = getElement('.article__form');
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js]: Використовуй класи, які використовуються тільки в js для селекторів і додавай но назви префікс js- . зміни в логіці мають по мінімуму впливати на стилі та розмітку

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.scss [hunk 21]
```diff
@@ -0,0 +1,223 @@
+$active_text_color: #FE7E11;
```
Comment (TArch64) [inline homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.scss]: Давай поки без препроцесорів. До того ж тяжко дивитись на ці всі вкладеності. І якщо тобі потрібні змінні, то використовуй css variables. Для цього сас вже не потрібен, а кількість вкладеностей в селекторах робить це все нечитабельним і тяжко зрозуміти що до чого належить і що для чого використовується

**Review by TArch64 — CHANGES_REQUESTED** [review]

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html (hunk 17 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: хотів пізніше доробити

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js (hunk 19 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.js:1]: Якщо треба буде змінити querySelector на щось інше то треба буде повсюду його заміняти)
Просто так красивіше

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html (hunk 18 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Ніколи з таким проблем не було)

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.scss (hunk 21 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.scss]: без препроцесору я б верстав набагато довше)
нажаль в мене немає стільки часу

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html (hunk 14 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Він там раніше був)

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html (hunk 16 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Ну тут версталось тільки для того щоб була форма на другу домашку.
Якщо що можу лишній section убрати

File: homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html (hunk 18 above)
Comment (YaroslavO) [reply homeworks/yaroslav.olenenko_YaroslavO/homework_2/index.html]: Насправді давно вже вручну не підключав файли. За мене це робить webpack

**Pull request closed without merging**